lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -67,8 +71,30 @@ def destroy
     output = powershell(powershell_args, script_content)
     Puppet.debug "Destroy Dsc Resource returned: #{output}"
     data = JSON.parse(output)
+
     fail(data['errormessage']) if !data['errormessage'].empty?
-    true
+
+    notify_reboot_pending if data['rebootrequired'] == true
+
+    data
+  end
+
+  def notify_reboot_pending
+    Puppet.info "A reboot is required to progress further. Notifying Puppet."
+
+    reboot_resource = resource.catalog.resource(:reboot, 'dsc_reboot')
+    unless reboot_resource
+      Puppet.warning "No reboot resource found in the graph that has 'dsc_reboot' as it's name. Cannot signal reboot to Puppet."
+      return
+    end
+
+    if reboot_resource.provider.respond_to?(:reboot_required)
+      # internal API used to let reboot resource knows a reboot is pending
+      reboot_resource.provider.reboot_required = true
+    else
+      Puppet.warning "Reboot module must be updated, since resource does not have :reboot_required method implemented. Cannot signal reboot to Puppet."
```

Comment:
Should we change the wording for this line? If the reboot resource is present, we do actually reboot the host.